### PR TITLE
Add a click-through note to remote users for certain requests

### LINKED ISSUE #551: Add IP check to mediated, in-library requests
This would apply to Spec, Rumsey, and Hoover Archives.

Can we do an IP check when the user clicks the Request button, and if they are outside the "campus + VPN" range, popup up an interim page, like so:

on-site-access-ip address check

IP ranges are in Consul
https://consul.stanford.edu/display/ERU/IP+Addresses+-+Default

### PR DESCRIPTION
**Pull request by jkeck**
Closes #551 

not-at-su

### CODE REVIEW

**Comment by ndushay** [comment]
LGTM;  👍

File: app/models/ip_range.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  class << self
+    def includes?(ip_to_check)
+      new(ip_to_check).included?
+    end
+
+    def default_network
+      Settings.stanford_ips
+    end
+  end
+
```
Comment (tingulfsen) [inline app/models/ip_range.rb]: After merging, I realized that I didn't quite understand your intent here. What were you trying to achieve with lines 33 through 41?

File: spec/models/ip_range_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  context 'ranges' do
+    it 'IP addresses configured as custom ranges are properly identified' do
+      expect(described_class.new('171.10.15.6', test_network)).not_to be_included
+      expect(described_class.new('171.10.15.7', test_network)).to be_included
+      expect(described_class.new('171.10.15.20', test_network)).to be_included
+      expect(described_class.new('171.10.15.35', test_network)).to be_included
+      expect(described_class.new('171.10.15.36', test_network)).not_to be_included
+    end
+
+    it 'IP addresses configured as ranges using slash suffix are properly identified' do
```
Comment (jmartin-sul) [inline spec/models/ip_range_spec.rb]: seems like a totally reasonable test, but does the slash suffix actually come into play on these?

File: app/models/ip_range.rb (hunk 1 above)
Comment (jkeck) [inline app/models/ip_range.rb]: They are two completely unrelated methods.  They are both providing class methods so we can say things like `self.class.default_network` as is in the initializer or `IPRange.includes?` as is used in the helper.

**Pull request merged**